crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -23,24 +24,21 @@ pub(crate) struct PubGrubPriorities(FxHashMap<PackageName, PubGrubPriority>);
 
 impl PubGrubPriorities {
     /// Add a [`PubGrubPackage`] to the priority map.
-    pub(crate) fn insert(&mut self, package: &PubGrubPackage, version: &Range<Version>) {
+    pub(crate) fn insert(
+        &mut self,
+        package: &PubGrubPackage,
+        version: &Range<Version>,
+        urls: &ForkUrls,
+    ) {
         let next = self.0.len();
         match &**package {
             PubGrubPackageInner::Root(_) => {}
             PubGrubPackageInner::Python(_) => {}
 
-            PubGrubPackageInner::Marker {
-                name, url: None, ..
-            }
-            | PubGrubPackageInner::Extra {
-                name, url: None, ..
-            }
-            | PubGrubPackageInner::Dev {
-                name, url: None, ..
-            }
-            | PubGrubPackageInner::Package {
-                name, url: None, ..
-            } => {
+            PubGrubPackageInner::Marker { name, .. }
+            | PubGrubPackageInner::Extra { name, .. }
+            | PubGrubPackageInner::Dev { name, .. }
+            | PubGrubPackageInner::Package { name, .. } => {
```

Comment:
https://github.com/astral-sh/uv/blob/153484bc181795b367917523cabd53fbf711526a/crates/uv-resolver/src/pubgrub/package.rs#L205-L217 is _almost_ helpful here, except it returns a name when `Root` has a name, which would be slightly different than the behavior here.

Reply:
What do you think about? We would also use this in `request_package`

```
    /// Returns the name of this PubGrub package, if it has one and is not the root package.
    pub(crate) fn name_no_root(&self) -> Option<&PackageName> {
        match &**self {
            PubGrubPackageInner::Root(_) | PubGrubPackageInner::Python(_) => None,
            PubGrubPackageInner::Package { name, .. }
            | PubGrubPackageInner::Extra { name, .. }
            | PubGrubPackageInner::Dev { name, .. }
            | PubGrubPackageInner::Marker { name, .. } => Some(name),
        }
    }
```

Reply:
(If yes, i'll make a follow-up PR, the stack is large enough)

Reply:
Yeah I think that's fine.